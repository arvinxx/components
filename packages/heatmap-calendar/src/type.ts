/**
 * 热力图数据结构
 */
export interface ItemData {
  /**
   * 日期
   */
  date: string;
  /**
   * 天数
   */
  day: number;
  /**
   * 月份
   */
  month: number;
  /**
   * 热点数
   */
  total: number;
  /**
   * 周
   */
  week: number;
  /**
   * 年
   */
  year: number;
}

export type HeatmapCalendarData = ItemData[];
