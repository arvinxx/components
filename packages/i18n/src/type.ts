import type { ComponentProps, FC } from 'react';
import type { IntlProvider } from 'react-intl';

export type LocalesType = 'en-US' | 'zh-CN';

export type LocaleKey = string;

export type IntlProviderProps<T = Record<string, string>> = FC<
  Omit<ComponentProps<typeof IntlProvider>, 'messages'> & {
    messages: T;
  }
>;
