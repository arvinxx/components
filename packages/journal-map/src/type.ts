/**
 * 阶段
 */
export interface Step<T = string> {
  id: T;
  name: string;
  /**
   * 该步骤的颜色
   */
  color?: string;
}

/**
 * 用户旅程地图数据
 */
export interface JournalMapData<T extends string = string> {
  /**
   * 大的步骤
   */
  steps: Step<T>[];
  /**
   * 用户行为
   */
  actions: Record<T, UserAction[]>;
}

/**
 * 用户行为
 */
export interface UserAction {
  /**
   * 行为名称
   */
  title: string;
  /**
   * 说明
   */
  description?: string;

  /**
   * 有关的想法
   */
  thoughts?: string[];
  /**
   * 情绪得分
   */
  emotion: number;
  /**
   * 痛点
   */
  painSpot?: string[];
}
