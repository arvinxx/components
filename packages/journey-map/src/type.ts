/**
 * 阶段
 */
export interface Step<T = string> {
  /**
   * 阶段id
   * @TJS-type string
   */
  id: T;
  /**
   * 阶段名称
   */
  name: string;
  /**
   * 该阶段的颜色
   */
  color?: string;
}

/**
 * 用户旅程地图数据
 */
export interface JourneyMapData<T extends string = string> {
  /**
   * 大的步骤
   */
  steps: Step<T>[];
  /**
   * 用户行为
   */
  actions: Record<T, UserAction[]>;
}

/**
 * 用户行为
 */
export interface UserAction {
  /**
   * 行为名称
   */
  title: string;
  /**
   * 说明
   */
  description?: string;

  /**
   * 有关的想法
   */
  thoughts?: string[];
  /**
   * 情绪得分
   */
  emotion: number;
  /**
   * 痛点
   */
  painSpot?: string[];
}

/**
 * 解析后的 YML 数据
 */
export interface JourneyMapYML<T extends string = string> {
  /**
   * 步骤
   */
  steps: Step<T>[];
  /**
   * 用户行为
   */
  actions: Record<T, Record<T, number>[]>;
}
