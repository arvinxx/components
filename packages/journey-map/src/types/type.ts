/**
 * 阶段
 */
export interface Stage<T = string> {
  /**
   * 阶段id
   * @TJS-type string
   */
  id: T;
  /**
   * 阶段名称
   */
  name: string;
  /**
   * 该阶段的颜色
   */
  color?: string;
}

/**
 * 用户旅程地图数据
 */
export interface JourneyMapData<T extends string = string> {
  /**
   * 大的步骤
   */
  stages: Stage<T>[];
  /**
   * 用户行为
   */
  actions: Record<T, UserAction[]>;
}

/**
 * 用户行为类型
 */
export interface UserAction {
  /**
   * 行为名称
   */
  name: string;
  /**
   * 情绪得分
   * @minimum -2
   * @maximum 2
   * @TJS-type integer
   */
  emotion?: number;

  /**
   * 说明
   */
  description?: string;

  /**
   * 想法
   */
  thoughts?: string[];

  /**
   * 痛点
   */
  painSpot?: string[];
}
