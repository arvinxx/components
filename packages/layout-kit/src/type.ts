export type ContentPosition =
  | 'center'
  | 'end'
  | 'flex-end'
  | 'flex-start'
  | 'start'
  | 'stretch';

export type ContentDistribution =
  | 'center'
  | 'space-around'
  | 'space-between'
  | 'space-evenly'
  | 'stretch';

export type FlexDirection =
  | 'vertical'
  | 'vertical-reverse'
  | 'horizontal'
  | 'horizontal-reverse';
