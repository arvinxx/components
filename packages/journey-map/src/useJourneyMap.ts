import useMergeValue from 'use-merge-value';
import { useEffect, useState } from 'react';

import { getServiceToken, YMLToJSON } from './utils';
import type { YMLJourneyMapData, JourneyMapData } from './types';

interface StoreParams {
  data: JourneyMapData | string;
  onChange?: (data: JourneyMapData) => void;
  title?: string;
}

export const useJourneyMap = ({
  data,
  onChange,
  title: initTitle,
}: StoreParams) => {
  const defaultJourneyMap: JourneyMapData = { stages: [], actions: {} };

  const hook = useMergeValue<JourneyMapData>(
    typeof data === 'string' ? defaultJourneyMap : data,
    { onChange },
  );

  const [title, setTitle] = useState<string>(initTitle);

  const [store, setStore] = hook;
  /**
   * 从 YML URL 中解析 JSON
   * @param url
   */
  const fetchYMLToJSON = async (url: string): Promise<YMLJourneyMapData> => {
    const res = await fetch(url);
    const yml = await res.text();

    return YMLToJSON(yml);
  };

  useEffect(() => {
    if (typeof data === 'string' && data.startsWith('http')) {
      fetchYMLToJSON(data).then((result) => {
        const { stages, actions } = result;
        setTitle(result.title);
        setStore({ stages, actions });
      });
    }
  }, [data]);

  const actionList = Object.values(store.actions).flat();

  return { store, actionList, actionLength: actionList.length, title };
};

export const JourneyMapStore = getServiceToken(useJourneyMap);
