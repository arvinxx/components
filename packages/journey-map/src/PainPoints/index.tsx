import type { FC } from 'react';
import React, { useContext } from 'react';

import { ListSection } from '../components';
import { JourneyMapStore } from '../useJourneyMap';

const PainPoints: FC = () => {
  const { config } = useContext(JourneyMapStore);
  const height = config?.height?.painPoint;

  return (
    <ListSection title={'痛点'} sectionKey={'painPoints'} height={height} />
  );
};

export default PainPoints;
