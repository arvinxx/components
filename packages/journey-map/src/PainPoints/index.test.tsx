import React from 'react';
import { render } from '@testing-library/react';

import PainPoints from './index';
import { JourneyMapStore, useJourneyMap } from '../useJourneyMap';

import type { FC } from 'react';
import type { UserAction } from '../types';

const Demo: FC<{ actions: Record<string, UserAction[]> }> = ({ actions }) => {
  const value = useJourneyMap({
    data: {
      actions,
      stages: [
        { id: '1', name: '1' },
        { id: '2', name: '2', color: 'blue' },
      ],
    },
  });
  return (
    <JourneyMapStore.Provider value={value}>
      <PainPoints />
    </JourneyMapStore.Provider>
  );
};

describe('Actions', () => {
  it('默认状态', () => {
    const { container } = render(
      <Demo
        actions={{
          '1': [{ name: '1', painPoints: ['1', '2'] }],
          '2': [{ name: '2', painPoints: ['2-1', '2-2'] }],
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
