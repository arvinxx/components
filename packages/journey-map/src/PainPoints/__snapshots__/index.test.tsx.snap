// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions 默认状态 1`] = `
<div>
  <div
    class="avx-journey-map-list-container"
  >
    <div
      class="avx-journey-map-section"
    >
      <div
        class="avx-journey-map-section-text"
      >
        痛点
      </div>
    </div>
    <div
      class="avx-journey-map-list-content"
    >
      <div
        class="avx-journey-map-list-card"
        style="width: 25%; margin-right: 25%; border-color: #bae7ff;"
      >
        <ul>
          <li>
            1
          </li>
          <li>
            2
          </li>
        </ul>
      </div>
      <div
        class="avx-journey-map-list-card"
        style="width: 25%; margin-left: 25%; border-color: blue;"
      >
        <ul>
          <li>
            2-1
          </li>
          <li>
            2-2
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
